ThunderDome/ThunderDome/monsterBlock.cs

```diff
@@ -57,91 +130,18 @@ public static string healthBar(monsterBlock monster)
             return $"HP: [{hpChunks}{hpEmpties}]";
         }
 
-        //Attack() method is called whenever a monsterBlock makes an attack
-        public static int Attack(monsterBlock attacker, monsterBlock target)
-        {
-            //Attack fails if either attacker or target is dead
-            if (target.isDead == true)
-            {
-                Console.WriteLine($"{target.name} is already dead!");
-                return 0;
-            }
-            else if (attacker.isDead == true)
-            {
-                Console.WriteLine($"{attacker.name} is already dead!");
-                return 0;
-            }
-            else
-            {
-                Random roll = new Random();
-                int attackRoll = roll.Next(1, 20);
-                int toHit = attackRoll + attacker.attackModifier;
-
-                //on a hit, calls takeDamage() method, to deal damage to the target
-                if (toHit >= target.armorClass)
-                {
-                    int damageRoll = roll.Next(1, 6);
-                    int damageDelt = damageRoll + attacker.damageModifier;
-                    takeDamage(damageDelt, target);
-                    Console.WriteLine($"{attacker.name} hit {target.name} for {damageDelt} points of damage!\n");
-                    //Console.WriteLine($"{attacker.name} dealt {damageDelt} points of damage to {target.name} \n");
-
-                }
-                else
-                {
-                    Console.WriteLine($"{attacker.name} missed on {target.name}!\n");
-                }
-                return toHit;
-            }
-
-        }
         //method takeDamage() reduces the target's currentHealth on a hit. 
-        public static int takeDamage(int damage, monsterBlock target)
+        public int takeDamage(int damage)
         {
-            target.currentHealth -= damage;
+            this.currentHealth -= damage;
```

Comment:
notice the use of `this`. You should cover this in class, if not ask.  But instead of passing the object into the method you simply refer to the current instance of the object as `this`